app/frontend/src/components/EditLocationMap.tsx

```diff
@@ -120,52 +121,96 @@ export default function EditLocationMap({
     map.current!.off("touchmove", moveHandler);
     map.current!.getCanvas().style.cursor = "move";
 
-    centerCoords.current = e.lngLat;
-    redrawMap();
+    centerCoords.current = e.lngLat.wrap();
     syncCoords();
+    redrawMap();
   };
 
   const redrawMap = () => {
+    if (!isBlank.current) {
+      map.current!.setLayoutProperty("circle", "visibility", "visible");
+    }
     if (!exact) {
       (map.current!.getSource("circle") as GeoJSONSource).setData(
-        circleGeoJson(centerCoords.current!, radius.current)
+        circleGeoJson(centerCoords.current, radius.current)
       );
     } else {
       (map.current!.getSource("circle") as GeoJSONSource).setData(
-        pointGeoJson(centerCoords.current!)
+        pointGeoJson(centerCoords.current)
       );
     }
   };
 
   // syncs imperative coordinates into the reactive location object
   const syncCoords = () => {
-    const wrapped = centerCoords.current!.wrap();
-    commit((location) => {
-      return {
-        ...location,
-        lat: wrapped.lat,
-        lng: wrapped.lng,
-      };
+    const wrapped = centerCoords.current;
+    commit({
+      lat: wrapped.lat,
+      lng: wrapped.lng,
     });
   };
 
-  const commit = (update: (s: ApproximateLocation) => ApproximateLocation) => {
-    setLocation(update);
-    updateLocation(location);
+  const commit = (
+    updates: Partial<ApproximateLocation>,
+    update: boolean = true
+  ) => {
+    const current = {
+      address: address.current,
+      lat: centerCoords.current.lat,
+      lng: centerCoords.current.lng,
+      radius: exact ? 0 : radius.current,
+    } as ApproximateLocation;
+    if ("address" in updates) {
```

Comment:
if it's not valid for these values to be [falsey](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), just do `if (updates.address) {`. That way you can remove all the non-null assertions

Reply:
thanks, again, I don't know how undefined/null/falsey work in js... :cry: